src/AppBundle/Controller/PredictionController.php

```
@@ -0,0 +1,91 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use AppBundle\Entity\Prediction;
+use AppBundle\Entity\Race;
+use AppBundle\Form\Type\VoteType;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+* Create the prediction overview and allow users to vote
+*
+* @author Maarten van Asperen <maarten@connectholland.nl>
+*/
+class PredictionController extends Controller
+{
+    /**
+    * An overview of all the races
+    *
+    * @return Response
+    */
+    public function homeAction(){
+        $entityRepository = $this->getDoctrine()->getRepository(Race::class);
+        $races = $entityRepository->findAll();
+
+        return $this->render(
+            'pages/predictions/home.html.twig',
+            [
+                'races' => $races,
+            ]
+        );
+    }
+
+    /**
+     * A detail action where the user can vote
+     *
+     * @param $id
+     * @param Request $request
+     *
+     * @return Response|RedirectResponse
+     */
+    public function detailAction($id, Request $request){
+        $prediction = new Prediction();
+
+        $race = $this->getDoctrine()->getRepository(Race::class)->findOneById($id);
+
+        $form = $this->createForm(VoteType::class, $prediction);
+        $form->handleRequest($request);
+
+        if ( $form->isSubmitted() && $form->isValid()) {
+            $prediction->setUser($this->getUser());
+            $prediction->setCorrect(false);
+            $prediction->setRace($race);
+
+            $em = $this->getDoctrine()->getManager();
+
+            $repository = $this->getDoctrine()->getRepository(Prediction::class);
+
+            $result = $repository->findOneBy([
+                'race' => $race,
+                'user' => $this->getUser()
+            ]);
+
+            if (empty($result) === false) {
+                die('je hebt al gestemd!');
```

You should throw an exception here